dcm4che-net-audit/src/test/java/org/dcm4che3/net/audit/AuditLoggerTest.java

```diff
@@ -0,0 +1,356 @@
+    }
+
+    @Test
+    public void reconfigure_UpdatesSpoolDirectory_WhenCalled() {
+        final File newDir = new File("var/local/test");
+        newLogger.setSpoolDirectory(newDir);
+        logger.setSpoolDirectory(new File(""));
+
+        logger.reconfigure(newLogger);
+
+        assertThat("spoolDirectory is not as expected", logger.getSpoolDirectory(), is(newDir));
+    }
+
+    @Test
+    public void reconfigure_UpdatesSpoolFileNamePrefix_WhenCalled() {
+        newLogger.setSpoolFileNamePrefix("PRE");
+        logger.setSpoolFileNamePrefix("POST");
+
+        logger.reconfigure(newLogger);
+
+        assertThat("spoolFileNamePrefix is not as expected", logger.getSpoolNameFilePrefix(), is("PRE"));
+    }
+
+    @Test
+    public void reconfigure_UpdatesSpoolFileNameSuffix_WhenCalled() {
+        newLogger.setSpoolFileNameSuffix("POST");
+        logger.setSpoolFileNameSuffix("PRE");
+
+        logger.reconfigure(newLogger);
+
+        assertThat("spoolFileNameSuffix is not as expected", logger.getSpoolFileNameSuffix(), is("POST"));
+    }
+
+    @Test
+    public void reconfigure_UpdatesRetryInterval_WhenCalled() {
+        newLogger.setRetryInterval(10);
+        logger.setRetryInterval(1);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("retryInterval is not as expected", logger.getRetryInterval(), is(10));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditLoggerInstalled_WhenCalled() {
+        newLogger.setAuditLoggerInstalled(false);
+        logger.setAuditLoggerInstalled(true);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("auditLoggerInstalled is not as expected", logger.getAuditLoggerInstalled(), is(false));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditRecordRepositoryDevices_WhenCalled() {
+        final Connection auditUDP = new Connection("audit-udp", "localhost", 514);
+        auditUDP.setProtocol(Connection.Protocol.SYSLOG_UDP);
+
+        final AuditRecordRepository arr1 = new AuditRecordRepository();
+        arr1.addConnection(auditUDP);
+
+        final Device arrDevice1 = new Device("syslog");
+        arrDevice1.addDeviceExtension(arr1);
+        arrDevice1.addConnection(auditUDP);
+
+        newLogger.setAuditRecordRepositoryDevices(Collections.singletonList(arrDevice1));
+
+        final Connection auditTCP = new Connection("audit-tcp", "localhost", 516);
+        auditTCP.setProtocol(Connection.Protocol.SYSLOG_TLS);
+
+        final AuditRecordRepository arr2 = new AuditRecordRepository();
+        arr2.addConnection(auditTCP);
+
+        final Device arrDevice2 = new Device("syslog2");
+        arrDevice2.addDeviceExtension(arr2);
+        arrDevice2.addConnection(auditTCP);
+
+        logger.setAuditRecordRepositoryDevices(Collections.singletonList(arrDevice2));
+
+        logger.reconfigure(newLogger);
+
+        final Device toDevice = logger.getAuditRecordRepositoryDevices().get(0);

```

Comment:
Why not just use getAuditRecordRepositoryDevices() and compare List directly? Matchers can do equalTo on lists with reference check and since setAuditRecordRepositoryDevices assigns same List is same object instance so that would be better check than just getting first item.